docs/index.rst

```diff
@@ -2149,6 +2149,77 @@ You will need to configure manually Foundry. Unfortunately, this may mean duplic
 
 .. _stories:
 
+In-memory behavior
+~~~~~~~~~~~~~~~~~~
+
+Foundry allows to use "in-memory" repositories in your factories. This is mainly useful for `DDD <https://en.wikipedia.org/wiki/Domain-driven_design>`_
+applications or with `hexagonal architecture <https://en.wikipedia.org/wiki/Hexagonal_architecture_(software)>`, where
+repositories in domain are usually interfaces for which main implementations are a Doctrine one. You can tell Foundry to
+use the "in-memory" version of these repositories.
+
+First, you need to create an "in-memory" version of your repository. This repository must implement the
+``Zenstruck\Foundry\InMemory\InMemoryRepository`` interface. You can use the trait
+``Zenstruck\Foundry\InMemory\InMemoryRepositoryTrait`` to help you with this:
+
+::
+
+    use App\Domain\Address\DomainAddressRepositoryInterface;
+    use Zenstruck\Foundry\InMemory\InMemoryRepository;
+    use Zenstruck\Foundry\InMemory\InMemoryRepositoryTrait;
+
+    /**
+     * @implements InMemoryRepository<Address>
+     */
+    final class InMemoryAddressRepository implements InMemoryRepository, DomainAddressRepositoryInterface
+    {
+        /** @use InMemoryRepositoryTrait<Address> */
+        use InMemoryRepositoryTrait;
+
+        public static function _class(): string
+        {
+            return Address::class;
+        }
+
+        // + all methods implementing "DomainAddressRepository"
+    }
+
+Then, the "in-memory" repository should be used in Symfony's container, as the main repository implementation:
+
+.. code-block:: yaml
+
+    # config/services.yaml
+    when@test:
+        services:
+            App\Domain\Address\DomainAddressRepositoryInterface: '@App\Infrastructure\Address\InMemoryAddressRepository'
```

Comment:
Doesn't this have an impact on non-`#[AsInMemoryTest]` tests?

Reply:
you are right, I was currently reviewing this, and came to the same conclusion. In my project where I use this, I'm using a `InMemoryKernel`

Reply:
I replaced it with a simple sentence, with no code, I'll see in the future how I improve this section.